src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -828,9 +824,38 @@ BsonDocument getResumeToken(final MongoSourceConfig sourceConfig) {
   @Override
   public void commitRecord(final SourceRecord record, final RecordMetadata metadata) {
     if (metadata == null) {
-      currentStatistics.filteredRecords(1);
+      currentStatistics.getRecordsFiltered().sample(1);
     } else {
-      currentStatistics.successfulRecords(1);
+      currentStatistics.getRecordsAcknowledged().sample(1);
+    }
+  }
+
+  private void mongoCommandSucceeded(final CommandSucceededEvent event) {
```

Comment:
I still think that the entire CommandListener implementation should be broken out into a separate class and unit tested, but I can see the argument against, as it's tightly coupled to MongoSourceTask.

Reply:
All stat invocations are now explicitly covered, including the getmore failure. I added stats checks to existing integration tests. The only one that wasn't already covered was `records-filtered` (a Kafka callback stat).